sklearn/inspection/partial_dependence.py

```diff
@@ -345,22 +352,30 @@ def partial_dependence(estimator, X, features, response_method='auto',
                 "With the 'recursion' method, the response_method must be "
                 "'decision_function'. Got {}.".format(response_method)
             )
+        msg = "'estimator' parameter must be a fitted estimator"
+        if isinstance(estimator, BaseGradientBoosting):
+            fitted_attribute = 'estimators_'
+        else:
+            fitted_attribute = 'n_iter_'
+        check_is_fitted(estimator, fitted_attribute, msg=msg)
 
-    n_features = X.shape[1]
-    features = np.asarray(features, dtype=np.int32, order='C').ravel()
-    if any(not (0 <= f < n_features) for f in features):
-        raise ValueError('all features must be in [0, %d]'
-                         % (n_features - 1))
+    features_indices = np.asarray(
+        _get_column_indices(X, features), dtype=np.int32, order='C'
```

Comment:
Do we really need to explicitly set the dtype and order here?

Reply:
It was how `features` was defined. Integer dtype makes sense because we need indices. I don't know why the order C was necessary but anyway this is a 1d array so C or F ordered should be fine

Reply:
right, sorry didn't saw the diff. Let's leave it this way then.